src/pages/DelegatoryValidator/UnstakeDialog.tsx

```diff
@@ -5,41 +5,106 @@ import {
   DialogContent,
   DialogTitle,
   FormHelperText,
+  IconButton,
   InputAdornment,
   OutlinedInput,
+  Stack,
+  Typography,
 } from "@mui/material";
-import React from "react";
-import ContentBox from "../../components/IndividualPageContent/ContentBox";
-import ContentRow from "../../components/IndividualPageContent/ContentRow";
+import React, {useState} from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import {grey} from "../../themes/colors/aptosColorPalette";
+import {Types} from "aptos";
+import {parseTimestamp, timestampDisplay} from "../utils";
 
 type isDialogOpen = {
   handleDialogClose: () => void;
   isDialogOpen: boolean;
+  accountResource?: Types.MoveResource | undefined;
 };
 
 export default function UnstakeDialog({
   handleDialogClose,
   isDialogOpen,
+  accountResource,
 }: isDialogOpen) {
+  const lockedUntilSecs = accountResource
+    ? BigInt((accountResource.data as any).locked_until_secs)
```

Comment:
`as any` is probably not the best practice in TypeScript. We can create a type for accountResource and `as AccountResourceData`. See `BlockResourceData` for example. Also I feel that I saw this logic somewhere (`lockedUntilSecs = ...`), I would prompt you to think if it makes sense to create a helper function or something.